docs/docs/spec.md

```diff
@@ -241,8 +241,6 @@ exceptions in the subsequent sections:
 
 -   an array component type
```

Comment:
I was originally worried about what rule covers the root array type. But that is covered by e.g. a parameter type. So in `void foo(int @Nullable [] p)` the annotation is recognized because it is on the parameter.

Reply:
I guess that it is a little unfortunate that we use the term ["type component"](https://github.com/jspecify/jspecify/wiki/type-usages#type-component) and then _also_ speak of "array component types." But yeah, I think you're right that this rule is intended to refer only to the types of the elements of the array.